### PR TITLE
[Docs] add `CONTRIBUTING.md`

### PR DESCRIPTION
**Pull request by uncommon-type**
## Description

This PR makes sure that `jsx-eslint` projects have standardised content for the `CONTRIBUTING.md` file. It creates a `CONTRIBUTING.md` file with information common across all `jsx-eslint` projects and explains the ways to get involved, setup/installation details as well as steps for submitting Pull Requests.

## Acceptance Criteria

- [x]  Info is included on ways to contribute
- [x]  Details are provided on how to set up dev environment
- [x]  Guidelines are given on how to submits pull requests
- [x] Table of contents is added with links to document headings

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|    | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|   | :link: Update dependencies |
|  ✓  | :scroll: Docs              |

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+
+# Contributing to `{project_name}`
```
Comment (ljharb) [inline CONTRIBUTING.md]: will this actually work? like, will github dynamically insert this in??

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+[Commit messages](#commit-messages)
+
+[Code of Conduct](#code-of-conduct)
+
+
+
+## Ways to contribute
+
+There are various ways to get involved, here are a few suggestions of things we'd love some help with.
```
Comment (ljharb) [inline CONTRIBUTING.md]:
```suggestion
There are various ways to get involved; here are a few suggestions of things we'd love some help with.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+#### Resolving existing issues
+
+ You can consider helping out with [issues](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) already requiring attention.
```
Comment (ljharb) [inline CONTRIBUTING.md]: if "project_name" is dynamic, can the URL of the project be dynamic also?

if not, maybe:
```suggestion
 You can consider helping out with [issues](./issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) already requiring attention.
```
?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  ```
+  git checkout -b issue1234
+  ```
+* Get all dependencies
+  ```
+  npm install
+  ```
+* Run the tests and the build
+  ```
+  npm test && npm run build
```
Comment (ljharb) [inline CONTRIBUTING.md]: i think maybe `npm test` already runs the build for you? if not, let's simplify these instructions and change jsx-a11y to do that :-)

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+```
+git checkout -b issue1234
+```
+
+Commit the changes to your branch
+
+```
+git add .
+git commit
```
Comment (ljharb) [inline CONTRIBUTING.md]:
```suggestion
git commit -a
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+```
+
+Before sending the pull request, make sure your code is running on the latest available code by rebasing onto the upstream source
+
+```
+git fetch upstream
```
Comment (ljharb) [inline CONTRIBUTING.md:77]: i think we're missing instructions to add the "upstream" remote (i usually call this "source", but "upstream" is fine too)

File: CONTRIBUTING.md (hunk 1 above)
Comment (uncommon-type) [reply CONTRIBUTING.md]: Not really - I've replaced that placeholder with `the project`

File: CONTRIBUTING.md (hunk 6 above)
Comment (uncommon-type) [reply CONTRIBUTING.md:77]: Done😄

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+[Commit messages](#commit-messages)
+
+
+## Ways to contribute
+
+There are various ways to get involved. Here are a few suggestions of things we'd love some help with.
+
+#### Resolving existing issues
+
+ You can consider helping out with [issues](./issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) already requiring attention.
```
Comment (ljharb) [inline CONTRIBUTING.md:26]: hm, this link may not work after all, but we can see once it's landed

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+> **Note:** Before filing an issue, please check both open and closed issues to see if the problem has already been reported.
+
+#### Documentation
+
+We are happy to welcome contributions from anyone willing to improve documentation by adding missing information or making it more consistent and coherent.
+
+
+## Dev Environment
+
+* Install [Node.js](https://nodejs.org/en/)
```
Comment (ljharb) [inline CONTRIBUTING.md]:
```suggestion
* Install [Node.js](https://nodejs.org/en/), preferably with [nvm](https://github.com/nvm-sh/nvm)
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+
+```
+git fetch upstream
+git rebase upstream/main
+```
+
+Verify your changes
+
+```
+npm run test
```
Comment (ljharb) [inline CONTRIBUTING.md]:
```suggestion
npm test
# or
npm run tests-only
npm run lint
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+    * `Robustness` for a semver-patch refactor that makes the package more robust against environment breakage (f.ex., modifying built-in JS methods)
+    * `Performance` for a code change that improves performance 
+    * `Refactor` for no observable changes (semver-patch)
+    * `Actions` for changes to Github Actions workflows only
+    * `Tests` for anything to do with testing that's not just updating dev deps
+    * `Eslint` for config and/or code changes that affect linting
+    * `Meta` for changes to project infra not covered by the above (changes to package.json, github config files, etc.)
+    * `Patch` for anything semver-patch not covered by the above   
+* Use the present tense and imperative mood when providing a description of what you did
+* If your PR addresses an issue, reference it in the body of the commit message
```
Comment (ljharb) [inline CONTRIBUTING.md]: it might be better to link to the gist; then we don't have to repeat this content and i only have one place to keep up to date.

**Review by ljharb — APPROVED** [review]
Seems pretty good, let's land it and see if the links work

**Pull request merged**